src/system/SystemPacketBuffer.cpp

```diff
@@ -333,14 +333,18 @@ PacketBuffer* PacketBuffer::Consume(uint16_t aConsumeLength)
 {
     PacketBuffer* lPacket = this;
 
-    while (lPacket != NULL && aConsumeLength > 0)
+    while (aConsumeLength > 0)
     {
         const uint16_t kLength = lPacket->DataLength();
 
         if (aConsumeLength >= kLength)
         {
             lPacket = PacketBuffer::FreeHead(lPacket);
             aConsumeLength -= kLength;
+            if (lPacket == NULL)
+            {
+                break;
+            }
```

Comment:
I think leaving the lPacket != NULL check in line 336 to prevent it from entering the while loop at all if it is NULL is simpler than adding the break, isn't it?

Reply:
+1. 

Additionally, `this` can be `NULL`:  if and when it is, in case of `virtual` functions, you'd expect to get a fault at the call site; in cases such as this, you'd be faulting in the entry into the loop.